RecoMET/METPUSubtraction/plugins/PFMETProducerMVA.cc

```diff
@@ -16,12 +16,12 @@ PFMETProducerMVA::PFMETProducerMVA(const edm::ParameterSet& cfg)
   srcPFCandidatesView_ = consumes<reco::CandidateView>(cfg.getParameter<edm::InputTag>("srcPFCandidates"));
   srcVertices_     = consumes<reco::VertexCollection>(cfg.getParameter<edm::InputTag>("srcVertices"));
   vInputTag srcLeptonsTags = cfg.getParameter<vInputTag>("srcLeptons");
+  consumes<double>(edm::InputTag("fixedGridRhoFastjetAll"));
```

Comment:
Hi Matthieu,

I think the problem is that L1FastjetCorrector.cc accesses rho (in line 68 of https://cmssdt.cern.ch/lxr/source/JetMETCorrections/Algorithms/src/L1FastjetCorrector.cc?v=CMSSW_7_6_3),
without making a corresponding consumes call, so the calling EDProducers need to make this consumes call.

Cheers,


Christian

On Jan 19, 2016, at 4:55 PM, mmarionncern wrote:

> In RecoMET/METPUSubtraction/plugins/PFMETProducerMVA.cc:
> 
> > @@ -16,12 +16,12 @@ PFMETProducerMVA::PFMETProducerMVA(const edm::ParameterSet& cfg)
> >    srcPFCandidatesView_ = consumes<reco::CandidateView>(cfg.getParameter<edm::InputTag>("srcPFCandidates"));
> >    srcVertices_     = consumes<reco::VertexCollection>(cfg.getParameter<edm::InputTag>("srcVertices"));
> >    vInputTag srcLeptonsTags = cfg.getParameter<vInputTag>("srcLeptons");
> > +  consumes<double>(edm::InputTag("fixedGridRhoFastjetAll"));
> 
> Why hardcoding it?
> 
> �
> Reply to this email directly or view it on GitHub.
>